dialoptions.go

```diff
@@ -41,21 +41,42 @@ func newFuncDialOption(f func(*dialOptions)) *funcDialOption {
 	}
 }
 
+type funcDialOptionWithErr struct {
+	funcWithErr func(*dialOptions) error
+}
+
+func (fdo *funcDialOptionWithErr) apply(do *dialOptions) error {
+	return fdo.funcWithErr(do)
+}
+
+func newFuncDialOptionWithErr(f func(*dialOptions) error) *funcDialOptionWithErr {
+	return &funcDialOptionWithErr{
+		funcWithErr: f,
+	}
+}
+
 // WithTransportCredentials returns a DialOption which configures a connection
 // level security credentials (e.g., TLS/SSL).
 func WithTransportCreds(privKey ed25519.PrivateKey, serverPubKey ed25519.PublicKey) DialOption {
-	return newFuncDialOption(func(o *dialOptions) {
-		pubs := credentials.NewPublicKeys(serverPubKey)
+	return newFuncDialOptionWithErr(func(o *dialOptions) error {
+		privKey, err := credentials.ValidPrivateKeyFromEd25519(privKey)
+		if err != nil {
+			return err
+		}
+
+		pubs, err := credentials.ValidPublicKeysFromEd25519(serverPubKey)
+		if err != nil {
+			return err
+		}
```

Comment:
Ah so changing the signature of `DialOption` is pretty disruptive. What do you think about just changing the signature of `WithTransportCreds` instead?
```suggestion
	privKey, err := credentials.ValidPrivateKeyFromEd25519(privKey)
	if err != nil {
		return nil, err
	}

	pubs, err := credentials.ValidPublicKeysFromEd25519(serverPubKey)
	if err != nil {
		return nil, err
	}
	// ...
	return newFuncDialOptionWithErr(func(o *dialOptions) error {
		// ...
```

Reply:
Is that change potentially more disruptive?

Every call to DialWithContext would have to be updated to

```go
option, err := wsrpc.WithTransportCreds(privKey, serverPubKey)
if err != nil {
    panic(err)
}
conn, err := wsrpc.DialWithContext(ctx, "127.0.0.1:1338", option, wsrpc.WithBlock())
```

which would affect mercury + feeds. 

Potentially better long term solution:  separate dial from client initialization like we do in Server. (i.e. `NewClient` and `DialWithContext`)

Reply:
Another option would be to introduce a new func and just deprecate this one without breaking anything.

Reply:
Eh, yeah this is probably better.